src/codegen/context.ts

```diff
@@ -1,11 +1,48 @@
 import { CompilerContext } from "../context";
+import { topologicalSort } from "../utils/utils";
 import { FuncAstFunctionDefinition, FuncAstAsmFunction } from "../func/syntax";
 
-type ContextValues = {
-    function: FuncAstFunctionDefinition | FuncAstAsmFunction;
-};
+/**
+ * An additional information on how to handle the function definition.
+ * TODO: Refactor: we need only the boolean `skip` field in WrittenFunction.
+ * XXX: Writer.ts: `Body.kind`
+ */
+export type BodyKind = "asm" | "skip" | "generic";
+
+/**
+ * Replicates the `ctx.context` parameter of the old backends Writer context.
+ * Basically, it tells in which file the context value should be located in the
+ * generated Func code.
+ *
+ * TODO: Should be refactored; `type` seems to be redundant
```

Comment:
TypeScript does type erasure upon compilation, so in runtime this may be needed :)